src/compiler/checker.ts

```diff
@@ -20964,6 +20985,25 @@ namespace ts {
                         flags |= ModifierFlags.Async;
                         lastAsync = modifier;
                         break;
+
+                case SyntaxKind.OverrideKeyword:
+                  if (flags & ModifierFlags.Override) {
+                    return grammarErrorOnNode(modifier, Diagnostics._0_modifier_already_seen, "override");
+                  }
+                  else if (flags & ModifierFlags.Static) {
+                    return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "static", "override");
+                  }
+                  else if (flags & ModifierFlags.Private) {
+                    return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "private", "override");
+                  }
+                  else if (flags & ModifierFlags.Abstract) {
+                    return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "abstract", "override");
+                  }
+                  else if (node.parent.kind === SyntaxKind.ModuleBlock || node.parent.kind === SyntaxKind.SourceFile) {

```

Comment:
this should follow the `abstract` modifier checks. `override` is only applicable on a method declaration in a class or a class expression.